### PR TITLE
Schedule for 1.22 release 

### PR DESCRIPTION
**Pull request by savitharaghunathan**
#### What type of PR is this:

/kind documentation

#### What this PR does / why we need it:

This PR has the schedule for the 1.22 release, to facilitate discussion and gather feedback.

Follow up:

- Update the TBDs with the actual dates

#### Which issue(s) this PR fixes:

None

#### Special notes for your reviewer:

None

/area release-team
/sig release
/assign @justaugustus @saschagrunert @jeremyrickard @hasheddan
/cc @annajung @divya-mohan0209 @erismaster @kikisdeliveryservice 
/cc @JamesLaverack @mkorbi @MonzElmasry @Pensu @PI-Victor @pmmalinov01 @guineveresaenger 

/hold for reviews

### CODE REVIEW

File: releases/release-1.22/README.md
Comment (liggitt) [inline releases/release-1.22/README.md:41]: Many people will be out for a long-weekend around July 4th... that is likely to make it hard to find reviewers for last minute crunch items. I'm happy for the answer to that to be "don't expect last minute crunch items to make it in" if that's what we want.

Comment (jeremyrickard) [reply releases/release-1.22/README.md:41]: I'm in favor of "don't expect last minute crunch items to make it in". We should probably make sure we communicate this when we send out comms around the release.

**Comment by guineveresaenger** [comment]
I am in general a bit concerned around release team well-being for the daily burndown to last 4 weeks. 

Perhaps, together with @liggit 's concern around the U.S. 4th of July holiday (yes that's a US-centric reference, but we're talking _reviewer_ availability here) we could push it back to July 13? Also, that's my birthday so I'm partial. 😛

File: releases/release-1.22/README.md
Comment (savitharaghunathan) [reply releases/release-1.22/README.md:41]: +1

**Review by justaugustus — CHANGES_REQUESTED** [review]
@savitharaghunathan -- Would you mind splitting this into two separate PRs?

Keep this one for the date-specific discussions (since there is already discussion here) and create a separate one with only:
- OWNERS
- release-team.md
- README.md (with no dates)

That way we can bootstrap OWNERS while people are working out the schedule.

**Comment by savitharaghunathan** [comment]
> I am in general a bit concerned around release team well-being for the daily burndown to last 4 weeks.

It is definitely a valid point. We can definitely experiment with the length of burndown meetings.

> Perhaps, together with @liggitt 's concern around the U.S. 4th of July holiday (yes that's a US-centric reference, but we're talking _reviewer_ availability here) we could push it back to July 13? Also, that's my birthday so I'm partial. 😛

All the 3 docs deadlines come after 'code freeze' deadline. So, if we try and move it out, it will affect other deadlines that follow it and puts pressure on sig-docs & RT docs team.

**Comment by justaugustus** [comment]
@savitharaghunathan -- https://github.com/kubernetes/sig-release/pull/1531 has merged, so this is good to rebase now.

File: releases/release-1.22/README.md
Comment (ehashman) [inline releases/release-1.22/README.md:40]: Can we please push this date back? We had previously discussed [doing enhancements freeze earlier](https://github.com/kubernetes/sig-release/issues/1496#issuecomment-807329101) in the release cycle, e.g. in week 2. Week 3 might make for sense for the longer release cycle.

Comment (ehashman) [reply releases/release-1.22/README.md:40]: Ref: this was discussed with and acked by [chairs and leads](https://docs.google.com/document/d/1Jio9rEtYxlBbntF8mRGmj6Q1JAdzZ9fTDo3ru1HK_LI/edit#heading=h.3zavw6ipqibo) in the March meetings as well as the [PRR team](https://docs.google.com/document/d/10QkXwiZfBL7wBKHHFslooXvGDKHA75jkkgiIXuKFryM/edit#heading=h.nxm6b7vlrocv) in the Mar. 24 meeting.

File: releases/release-1.22/README.md
Comment (msau42) [reply releases/release-1.22/README.md:41]: In the past we've had code freeze before major holidays. I find that helps avoid scenarios where people end up working over the holidays.

Comment (pmmalinov01) [reply releases/release-1.22/README.md:41]: +1

File: releases/release-1.22/README.md
Comment (JamesLaverack) [reply releases/release-1.22/README.md:40]: Putting enhancements freeze on week three would put it on the 11th of May right? But the whole of the week before is KubeCon. I'm assuming reviewers are going to be hard to come by during that week.

If so does that mean that realisticly if you want a KEP reviewed for 1.22 you've got next week (week 1, April 26–30th) and the two days before enhancements freeze (week 2, May 10–11th). Is that too tight?

File: releases/release-1.22/README.md
Comment (soltysh) [inline releases/release-1.22/README.md:71]: This was raised before and I'll raise that again. It would be good to have beta.2 release a couple of days after code freeze, when all PRs merge. This approach allows downstream consumers to use the release payload sooner and provide feedback about the overall state of the release. I'd really like to see that being called out explicitly in the timeline.

File: releases/release-1.22/README.md
Comment (ehashman) [reply releases/release-1.22/README.md:40]: Realistically, most SIGs I'm involved in have already started 1.22 planning. In a 4/yr release cadence, enhancements freeze likely would have been next week or the week after. Week 3 doesn't seem that bad to me, particularly if we do it later in the week (is there any reason it has to be M/Tu? Is Thu an option?) I would say May 17/18th at the very latest.

File: releases/release-1.22/README.md
Comment (ehashman) [reply releases/release-1.22/README.md:41]: Thinking out loud: before July 4th implies a code freeze the week of June 28th, which is week 10. This would lead to a 5 week freeze/over a month frozen. This seems very long?

Why not do code freeze week 12 and test freeze week 13?

Comment (savitharaghunathan) [reply releases/release-1.22/README.md:41]: If we have the code freeze in week 12, docs deadlines will be pushed. All the 3 docs deadlines need an average of 2.5-3 weeks to be completed and that will be very close to the release.
I am in favor of communicating well in advance about the code freeze and having at least 1 week between the major deadlines and release day.

Comment (guineveresaenger) [reply releases/release-1.22/README.md:41]: I think it's somewhat ok for the docs deadlines to be pulled forward by a half week or so to make room for a shorter code freeze. See also on the main conversation: https://github.com/kubernetes/sig-release/pull/1530#issuecomment-824288017

Comment (kikisdeliveryservice) [reply releases/release-1.22/README.md:41]: q: wouldn't code freeze in week 10 be the same as a 4/year release? if the idea is to give people more time and breathing room it seems like adding time there somehow would be welcome/ a value-add for authors/sigs.

File: releases/release-1.22/README.md
Comment (savitharaghunathan) [reply releases/release-1.22/README.md:40]: @ehashman: what are your thoughts on moving enhancements freeze to May 13th? would it be a burden for PRR reviewers?

File: releases/release-1.22/README.md
Comment (logicalhan) [reply releases/release-1.22/README.md:41]: +1 on pushing the code freeze back to week 12, I would personally appreciate the breathing room.

File: releases/release-1.22/README.md
Comment (annajung) [reply releases/release-1.22/README.md:40]: Not a blocker comment, just want to point out that this will not only affect PRR reviewers but also the Enhancements team, especially the shadows. Ramping up and reviewing ~60+ Keps (an estimated number based on the last release), is going to put a lot of pressure on the team. Just something you should let your shadows know @JamesLaverack :)

File: releases/release-1.22/README.md
Comment (annajung) [reply releases/release-1.22/README.md:41]: I don't think the docs placeholder deadline should come before the code freeze, especially with how responsive everyone was to the docs team when the deadline came after the code freeze during the last release. If pushing code freeze, I think it might be good to push placeholder to the same day as code freeze or a few days after code freeze.

File: releases/release-1.22/README.md
Comment (puerco) [reply releases/release-1.22/README.md:71]: Let's do it. Just one detail:

The schedule we put together for 1.21 was tight timewise and would have been hard to squeeze another beta in. Some of the dates for 1.22 are still TBD but if we end up in the same situation where the weeks are too narrow we could perhaps decouple code freeze from beta.1. We freeze the branch, wait about a week, _and then cut beta.1_.  I don't know if there is a problem with that strategy I'm not seeing.

OTOH If the weeks around that time allow for it, we can cut a `beta.2`, no problem.

File: releases/release-1.22/README.md
Comment (PI-Victor) [reply releases/release-1.22/README.md:41]: > All the 3 docs deadlines need an average of 2.5-3 weeks to be completed and that will be very close to the release
This is indeed a concern to docs, with a longer release, we are definitely going to have more KEPs to track and if the 1.22 release cycle proves anything, although things went pretty good, is that people leave the docs PR last and are hard to get a hold of. Not only to submit the k/website PRs but also to get someone to do the tech review from their own SIG.
It would be ideal and it would speed the process up if someone commits to doing the technical docs review from the SIG, but i'm not sure if such a process is in place currently.

File: releases/release-1.22/README.md
Comment (savitharaghunathan) [reply releases/release-1.22/README.md]: The enhancements freeze has been updated to May 13th for now.

File: releases/release-1.22/README.md
Comment (savitharaghunathan) [reply releases/release-1.22/README.md]: Code freeze has been updated to July 8th and the deadline to open a docs placeholder PR will be July 9th. Test freeze has been updated to July 15 (one week after code freeze)

File: releases/release-1.22/README.md
Comment (savitharaghunathan) [reply releases/release-1.22/README.md:72]: @soltysh: added a TBD deadline for the beta.2 release cut

Comment (palnabarun) [reply releases/release-1.22/README.md:72]: @soltysh -- Thank you for bringing this up here as well. We did discuss it during the 1.21 retrospective and found this to be adding a lot of value. 👍🏽 

We had decided to put in a release cut (can be called beta.2 or rc.0, whatever is suitable) a few days post-Code Freeze.

@puerco -- We can ponder on the idea of decoupling Code Freeze from beta.1. Perhaps, maybe cut beta.1 the day after Code Freeze is put in?

File: releases/release-1.22/OWNERS
Comment (annajung) [inline releases/release-1.22/OWNERS:1]: I don't think this file should be deleted?

File: releases/release-1.22/release-team.md
Comment (annajung) [inline releases/release-1.22/release-team.md:1]: same here, I don't think this file should be deleted?

Comment (savitharaghunathan) [reply releases/release-1.22/release-team.md:1]: Good catch. Looking into it

File: releases/release-1.22/OWNERS
Comment (savitharaghunathan) [reply releases/release-1.22/OWNERS:1]: Good catch. Looking into it

Comment (savitharaghunathan) [reply releases/release-1.22/OWNERS]: Resolved it - https://github.com/savitharaghunathan/sig-release/tree/bootstrap-release-1.22/releases/release-1.22

File: releases/release-1.22/release-team.md
Comment (savitharaghunathan) [reply releases/release-1.22/release-team.md]: Resolved it - https://github.com/savitharaghunathan/sig-release/tree/bootstrap-release-1.22/releases/release-1.22

File: releases/release-1.22/README.md
Comment (sftim) [inline releases/release-1.22/README.md:44]: LGTM

File: releases/release-1.22/README.md
Comment (annajung) [inline releases/release-1.22/README.md:43]: July 27th is a Tuesday
```suggestion
- **Wednesday, July 28th**: Week 14 - Docs must be completed and reviewed
```

File: releases/release-1.22/README.md
Comment (annajung) [inline releases/release-1.22/README.md:52]:
```suggestion
| Start Enhancements Tracking | Enhancements Lead | Tue April 27 | week 1 | |
```

File: releases/release-1.22/README.md
Comment (savitharaghunathan) [reply releases/release-1.22/README.md:43]: Thanks for the catch!

File: releases/release-1.22/README.md
Comment (savitharaghunathan) [reply releases/release-1.22/README.md:52]: It is intended to start on the 26th. Will change the day to reflect the same

File: releases/release-1.22/README.md
Comment (savitharaghunathan) [reply releases/release-1.22/README.md]: done

**Comment by annajung** [comment]
Thanks Savitha! 
/lgtm

**Pull request merged**
